### ISSUE TITLE
Improve data/aws_ssm_parameter error message

### ISSUE THREAD

**Issue by ZsoltPath** [terraform-providers/terraform-provider-aws#15095]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently when the SSM parameter doesn't exist we only got a general error message:
```
Error: Error describing SSM parameter: ParameterNotFound:
```

It would be nice to include in the message the resource id and/or the missing parameter name.
For example:
```
data "aws_ssm_parameter" "test_parameter " {
  name  = "/dev/application/test"
}

Error: Error describing SSM parameter: test_parameter ParameterNotFound: '/dev/application/test'
```

### New or Affected Resource(s)

* data/aws_ssm_parameter

**Comment by bflad** [comment]
This would certainly be pretty helpful! If you or anyone is interested in adjusting this, this line (https://github.com/terraform-providers/terraform-provider-aws/blob/2dbd3ddeaf97e59d4cc4096257b1f87bd2f78b2e/aws/data_source_aws_ssm_parameter.go#L62) can be adjusted to:

```go
return fmt.Errorf("error describing SSM Parameter (%s): %w", name, err)
```

**Comment by bflad** [comment]
Adding the parameter name in the error message has been merged and will release with version 3.7.0 of the Terraform AWS Provider, later this week. Thanks to @ZsoltPath for the implementation. 👍
